mlflow/gateway/providers/togetherai.py

```diff
@@ -0,0 +1,470 @@
+            #":[DONE]
+
+        return completions.StreamResponsePayload(
+            id=resp.get("id"),
+            created=resp.get("created"),
+            model=config.model.name,
+            choices=[
+                completions.StreamChoice(
+                    index=idx,
+                    #TODO this is questionable since the finish reason comes from togetherai api
+                    finish_reason=choice.get("finish_reason"),
+                    delta=completions.StreamDelta(
+                        role=None,
+                        content=choice.get("text")
+                    )
+                )
+
+                for idx, choice in enumerate(resp.get("choices", []))
+            ],
+            # usage is not included in OpenAI StreamResponsePayload
+            #usage=completions.CompletionsUsage(
+            #    prompt_tokens=usage.get("prompt_tokens") if usage else None,
+            #    input_tokens=usage.get("completion_tokens") if usage else None,
+            #    total_tokens=usage.get("total_tokens") if usage else None,
+            #)
+        )
+
+    @classmethod
+    def completions_to_model(cls, payload, config):
+        key_mapping = {
+            # repetition_penalty (-3.402823669209385e+38, 3.402823669209385e+38)
+            # frequency_penalty (-2.0, 2.0)
+            # presence_penalty (-2.0, 2.0)
+            "frequency_penalty": "repetition_penalty",
+            # top_logprobs (0, 20)
+            # logprobs (-2147483648,2147483648)
+            "top_logprobs": "logprobs"
+        }
+
+        payload = rename_payload_keys(payload, key_mapping)
+
+        if "logprobs" in payload and payload["logprobs"] and "logprobs" not in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Missing paramater in payload."
+                "If flag logprobs parameter is set to True"
+                "then logsprobs parameter must contain a value.")
+            )
+
+        # TODO maybe let the api service handle these internally?
+
+        if "logitbias" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Invalid parameter in payload."
+                "Parameter logitbias is not supported for togetherai.")
+            )
+
+        if "seed" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Invalid parameter in payload."
+                "Parameter seed is not suuported of togetherai.")
+            )
+
+        if "presence_penalty" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail=("Invalid parameter in payload."
+                "Parameter presence_penalty is not supported for togetherai.")
+            )
```

Comment:
I think we don't need to validate all query parameters, unless they are listed as [standard parameters](https://mlflow.org/docs/latest/llms/deployments/index.html#completions). Users can pass any parameters after all, so we can let TogetherAI APIs to handle them and raise error.

Reply:
This makes total sense to me! On it!